### ISSUE TITLE
Selecting "already have DOI" but not passing one should be an error

### ISSUE THREAD

**Issue by fenekku** [inveniosoftware/react-invenio-deposit#293]
Package version (if known): v4.0+ (still valid in v6.0)

On the uploads page, selecting the "already have a DOI radio button", but not providing one doesn't result in an error but it should.

Steps:

1. Go to new upload
2. Select "Yes" to "Do you already have a DOI for this upload?" but don't fill anything in the textbox
3. Save & Publish without errors!
4. Notice that a new DOI has been assigned.

**Comment by ppanero** [comment]
This is a bit tricky:

DOIs are configured as system managed and mandatory in the service config. This means that if not present the system will create one.

1. The UI is saving the record without `pids: {}`. However, the backend responds with it (the `ExternalPIDsComponent` will add it).
2. The `ExternalPIDsComponent` would also add `pids: {}` if not present, so the point above is **not** the main problem but would also add the empty dictionary that would lead to the same situation.
3. The `ExternalPIDsComponent` is configured to add a DOI (or any other mandatory/system managed pid) **at all costs**.  It would only fail if a value is given but it is not valid.

TL;DR:

We need frontend validation to avoid making the request or we need to revisit the whole logistics of when to create a PID and when not to.

Removing that **at all costs** would result in "How do we tell from the REST API that we want a DOI/PID to be assigned?"

**Comment by ppanero** [comment]
Tackled in: https://github.com/inveniosoftware/invenio-rdm-records/pull/814
As per definition of https://codimd.web.cern.ch/PWRdWnQEQd6XLZnrrj2HLQ the UI (WIP) should send a payload such as:
```
doi:
   provider: external
   identifier: 
```

with no identifier value in that case (since the user did not input anything _yet_). This would result in a validation error.

This test https://github.com/inveniosoftware/invenio-rdm-records/blob/9e88da5043c122e0558d3636add103f122b48f8f/tests/services/pids/test_pids_service.py#L285 validates the implementation at backend (REST API) level... UI is WIP still...
